packages/siop-oid4vp/lib/authorization-request/Payload.ts

```diff
@@ -18,20 +18,28 @@ import { createRequestRegistration } from './RequestRegistration'
 import { ClaimPayloadOptsVID1, CreateAuthorizationRequestOpts, PropertyTarget } from './types'
 
 export const createPresentationDefinitionClaimsProperties = (opts: ClaimPayloadOptsVID1): ClaimPayloadVID1 => {
-  if (!opts || !opts.vp_token || (!opts.vp_token.presentation_definition && !opts.vp_token.presentation_definition_uri)) {
+  if (
+    !opts ||
+    !opts.vp_token ||
+    (!opts.vp_token.presentation_definition && !opts.vp_token.presentation_definition_uri && !opts.vp_token.dcql_query)
+  ) {
     return undefined
   }
-  const discoveryResult = PEX.definitionVersionDiscovery(opts.vp_token.presentation_definition)
-  if (discoveryResult.error) {
-    throw new Error(SIOPErrors.REQUEST_CLAIMS_PRESENTATION_DEFINITION_NOT_VALID)
+
+  if (opts.vp_token.presentation_definition || opts.vp_token.presentation_definition_uri) {
+    const discoveryResult = PEX.definitionVersionDiscovery(opts.vp_token.presentation_definition)
+    if (discoveryResult.error) {
+      throw new Error(SIOPErrors.REQUEST_CLAIMS_PRESENTATION_DEFINITION_NOT_VALID)
+    }
   }
 
   return {
     ...(opts.id_token ? { id_token: opts.id_token } : {}),
-    ...((opts.vp_token.presentation_definition || opts.vp_token.presentation_definition_uri) && {
+    ...((opts.vp_token.presentation_definition || opts.vp_token.presentation_definition_uri || opts.vp_token.dcql_query) && {
       vp_token: {
         ...(!opts.vp_token.presentation_definition_uri && { presentation_definition: opts.vp_token.presentation_definition }),
         ...(opts.vp_token.presentation_definition_uri && { presentation_definition_uri: opts.vp_token.presentation_definition_uri }),
+        ...(opts.vp_token.dcql_query && { dcql_query: opts.vp_token.dcql_query }),
```

Comment:
This is an error. Are you also sure you merged in latest PR/work from Animo to begin with? I remember making this same remark. The dcql_query needs to be brought outisde of the line 38. As the first line in the vp_token part (33-40) would also be true (!vp_token_definition_uri) in case of dcql, which would mean it would include an undefined presentation_definition all of a sudden.

So simply make a distinction between dcql and PD earlier on and based on that return the previous code/result or the dcql result.

Reply:
DCQL and PD are now separated